crates/provider/src/layers/gas.rs

```diff
@@ -144,30 +144,20 @@ where
         &'a self,
         tx: &'b mut N::TransactionRequest,
     ) -> Result<&'b mut N::TransactionRequest, TransportError> {
-        let gas_price = self.get_gas_price();
-
-        if tx.gas_limit().is_none() {
-            let gas_estimate = self.get_gas_estimate(tx);
-
-            match futures::join!(gas_price, gas_estimate) {
-                (Ok(gas_price), Ok(gas_estimate)) => {
-                    tx.set_gas_price(gas_price);
-                    tx.set_gas_limit(gas_estimate);
-                    Ok(tx)
-                }
-                (Ok(_gas_price), Err(err)) => Err(err),
-                (Err(err), Ok(_gas_estimate)) => Err(err),
-                (Err(err1), Err(_err2)) => Err(err1),
-            }
+        let gas_price_fut = self.get_gas_price();
+        let gas_limit_fut = if let Some(gas_limit) = tx.gas_limit() {
+            async move { Ok(gas_limit) }.left_future()
         } else {
-            let gas_price = gas_price.await;
-            match gas_price {
-                Ok(gas_price) => {
-                    tx.set_gas_price(gas_price);
-                    Ok(tx)
-                }
-                Err(err) => Err(err),
+            async { self.get_gas_estimate(tx).await }.right_future()
+        };
+
+        match futures::try_join!(gas_price_fut, gas_limit_fut) {
```

Comment:
this `match` block is equivalent to a `.map` right?

Reply:
Oh yes!